### PR TITLE
Fix documented type for entry point function

### PR DESCRIPTION
**Pull request by mwkroening**
What do you think?

### CODE REVIEW

File: src/boot_info.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use core::{ops, slice};
 /// ```
 /// # use bootloader::BootInfo;
 /// # type _SIGNATURE =
-/// extern "C" fn(boot_info: &'static BootInfo) -> !;
+/// fn(boot_info: &'static mut BootInfo) -> !;
```
Comment (phil-opp) [inline src/boot_info.rs:11]: The `extern "C"` is still part of the signature, even if it's not needed with the `entry_point` macro. So I think we should keep it here and instead add to the docs of `entry_point!` that it is not needed when using the macro.

**Review by phil-opp — COMMENTED** [review]
Good catch, thanks a lot!

File: src/boot_info.rs (hunk 1 above)
Comment (mwkroening) [reply src/boot_info.rs:11]: You are totally correct, thanks!

The docs of the macro don't mention `extern "C"`, so I think it's fine. :thinking:

**Comment by phil-opp** [comment]
Thanks!

**Pull request merged**